gcloud-java-compute/src/main/java/com/google/gcloud/compute/ImageConfiguration.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.api.services.compute.model.Image;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Base class for Google Compute Engine image configuration. Use {@link DiskImageConfiguration} to
+ * create an image from an existing Google Compute Engine disk. Use {@link RawImageConfiguration} to
+ * create an image from a file stored in Google Cloud Storage.
+ */
+public abstract class ImageConfiguration implements Serializable {
+
+  private static final long serialVersionUID = -9154332316597745316L;
+
+  private final SourceType sourceType;
+
+  /**
+   * Image source type. The only admissible value is {@code RAW}.
+   */
+  public enum SourceType {
+    RAW
+  }
+
+  /**
+   * Base builder for image configurations.
+   *
+   * @param <T> the image configuration class
+   * @param <B> the image configuration builder
+   */
+  public abstract static class Builder<T extends ImageConfiguration, B extends Builder<T, B>> {
+
+    private SourceType sourceType;
+
+    Builder() {}
+
+    Builder(ImageConfiguration imageConfiguration) {
+      this.sourceType = imageConfiguration.sourceType;
+    }
+
+    Builder(Image imagePb) {
+      if (imagePb.getSourceType() != null) {
+        this.sourceType = SourceType.valueOf(imagePb.getSourceType());
+      }
+    }
+
+    @SuppressWarnings("unchecked")
+    B self() {
+      return (B) this;
+    }
+
+    B sourceType(SourceType sourceType) {
```

Comment:
Is this something the user will need to set (should it be public or is it meant to be package scope?)